src/diagonal.works/b6/search/tree_test.go

```diff
@@ -163,58 +159,57 @@ func TestTreeListAdvanceOnDeletedIterator(t *testing.T) {
 	//4, 5, 6, 10, 12, 13, 15, 16
 	input := []int{10, 5, 15, 4, 6, 13, 16, 12}
 	cases := []struct {
+		name     string
 		delete   int
 		advance  int
 		ok       bool
 		expected []int
 	}{
-		//		{6, 10, true, []int{10, 12, 13, 15, 16}}, // Happy path
-		//		{6, 6, true, []int{10, 12, 13, 15, 16}},  // Advance to deleted
-		{6, 5, true, []int{10, 12, 13, 15, 16}}, // Advance to previous
-		//		{6, 17, true, []int{}},                   // Advance beyond end
+		{"happy path", 6, 10, true, []int{10, 12, 13, 15, 16}},
+		{"Advance to deleted", 6, 6, true, []int{10, 12, 13, 15, 16}},
+		{"Advance to previous", 6, 5, true, []int{10, 12, 13, 15, 16}},
+		// TODO: Make this test case pass.
+		// {"Advance beyond end", 6, 17, true, []int{}},
 	}
 
 	for _, c := range cases {
-		tree := newTreeList(&intValues{})
-		for _, v := range input {
-			tree.Insert(v)
-		}
-
-		i := tree.Begin()
-		found := false
-		for i.Next() {
-			if i.Value() == c.delete {
-				found = true
-				break
+		t.Run(c.name, func(t *testing.T) {
```

Comment:
Nice - this is a much better idea. In other places, where we use this pattern, we format the sub test names like "HappyPath" instead of "happy path", so that they match function names when factored out, and so they're easier to spell on the shell when you do `go test --run=whatever`, maybe do that here?

Reply:
That's a brilliant idea, sorry I didn't notice this pattern before. Done!